### PR TITLE
Revamp URL form

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Update the internal/external URL form to help guide users to fix their setup.

image

image

image

image

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by zsarnett — DISMISSED** [review]

File: src/panels/config/core/ha-config-url-form.ts [hunk 1]
```diff
@@ -43,46 +85,143 @@ class ConfigUrlForm extends LitElement {
...
+                    )}
+                  </div>
+                  <ha-formfield
+                    .label=${this.hass.localize(
+                      "ui.panel.config.url.external_use_ha_cloud"
+                    )}
+                  >
+                    <ha-switch
+                      .disabled=${disabled}
+                      .checked=${hasCloud && externalUrl === null}
```
Comment (ludeeus) [inline src/panels/config/core/ha-config-url-form.ts:108]:
```suggestion
                      .checked=${externalUrl === null}
```
You only get here if `hasCloud` is true

File: src/panels/config/core/ha-config-url-form.ts [hunk 2]
```diff
@@ -43,46 +85,143 @@ class ConfigUrlForm extends LitElement {
...
-              )}
-              name="internal_url"
-              type="url"
-              .disabled=${disabled}
-              .value=${this._internalUrlValue}
-              @value-changed=${this._handleChange}
-            >
-            </paper-input>
+
+            <ha-formfield label="Automatic">
```
Comment (ludeeus) [inline src/panels/config/core/ha-config-url-form.ts:176]: localize?

**Review by ludeeus — DISMISSED** [review]

File: src/translations/en.json [hunk 3]
```diff
@@ -1355,8 +1355,8 @@
                 "metric_example": "Celsius, kilograms",
                 "find_currency_value": "Find your value",
                 "save_button": "Save",
-                "external_url": "Internet",
-                "internal_url": "Local network",
+                "external_url": "External URL",
+                "internal_url": "Internal URL",
```
Comment (zsarnett) [inline src/translations/en.json:1359]: ?

Comment (balloob) [reply src/translations/en.json:1359]: They are not used in that form, or probably anywhere else.

**Review by zsarnett — APPROVED** [review]

**Pull request merged**